features/285-design-webhooks/use-cases.md

```diff
@@ -0,0 +1,285 @@
+# Sample Use Cases
+
+
+## Synchronize Metadata to Self-Managed Profile
+
+Developer may want to synchronize metadata when it is updated by user:
+- When user update metadata, an external service should be notified to
+  synchronize their managed profile according to user metadata in auth gear.
+- External service may want to validate & reject invalid user metadata.
+
+### Suggested Solution
+
+- In `before_user_metadata_update`:
```

Comment:
From events.md, it should be `before_user_update`?

Reply:
Yes it should be.